data_steward/cdr_cleaner/cleaning_rules/ehr_submission_data_cutoff.py

```diff
@@ -0,0 +1,240 @@
+
+            date_fields = []
+            datetime_fields = []
+
+            for field in fields:
+                # appends only the date columns to the date_fields list
+                if field['type'] in ['date']:
+                    date_fields.append(field['name'])
+
+                # appends only the datetime columns to the datetime_fields list
+                if field['type'] in ['timestamp']:
+                    datetime_fields.append(field['name'])
+
+            # will render the queries only if a CDM table contains a date or datetime field
+            # will ignore the CDM tables that do not have a date or datetime field
+            if date_fields or datetime_fields:
+                sandbox_query = {
+                    cdr_consts.QUERY:
+                        SANDBOX_QUERY.render(
+                            project_id=self.project_id,
+                            sandbox_id=self.sandbox_dataset_id,
+                            intermediary_table=self.sandbox_table_for(table),
+                            dataset_id=self.dataset_id,
+                            cdm_table=table,
+                            date_fields=(", ".join(date_fields)),
+                            datetime_fields=(", ".join(datetime_fields)),
+                            cutoff_date=self.cutoff_date),
+                }
+
+                sandbox_queries_list.append(sandbox_query)
+
+                date_cutoff_query = {
+                    cdr_consts.QUERY:
+                        DATE_CUTOFF_QUERY.render(
+                            project_id=self.project_id,
+                            dataset_id=self.dataset_id,
+                            cdm_table=table,
+                            sandbox_id=self.sandbox_dataset_id,
+                            intermediary_table=self.sandbox_table_for(table)),
+                    cdr_consts.DESTINATION_TABLE:
+                        table,
+                    cdr_consts.DESTINATION_DATASET:
+                        self.dataset_id,
+                    cdr_consts.DISPOSITION:
+                        bq_consts.WRITE_TRUNCATE
+                }
+
+                queries_list.append(date_cutoff_query)
+
+        return sandbox_queries_list + queries_list
+
+    def setup_rule(self, client):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
+
+    def setup_validation(self, client):
+        """
+        Run required steps for validation setup
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def validate_rule(self, client):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def get_sandbox_tablenames(self):
+        sandbox_tables = []
+        for table in self.affected_tables:
+            sandbox_tables.append(f'{self._issue_numbers[0].lower()}_{table}')
```

Comment:
You could use the `sandbox_table_for` helper function here, I think.
```suggestion
            sandbox_tables.append(self.sandbox_table_for(table))
```

Reply:
I changed it and both the unit and integration tests passed 👍🏾